core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1873,21 +1902,19 @@ struct mobj_ffa *thread_spmc_populate_mobj_from_rx(uint64_t cookie)
 	struct mobj_ffa *mf = NULL;
 	unsigned int num_pages = 0;
 	unsigned int offs = 0;
+	unsigned int range_count = 0;
+	uint32_t length = 0;
+	uint32_t frag_length = 0;
+	uint64_t handle = 0;
+	unsigned int idx = 0;
 	void *buf = NULL;
 	struct thread_smc_args ffa_rx_release_args = {
 		.a0 = FFA_RX_RELEASE
```

Comment:
Maybe FFA_RX_RELEASE should be submitted after every FFA_MEM_FRAG_RX or FFA_MEM_RETRIEVE_REQ_32,
or rx buffer is full in spmc perspective

Reply:
rx buffer is full in spmc perspective, what happens if I release rx_buffer after I select the complete retrieve?

Reply:
@jenswi-linaro Do you have any good suggestions？

Reply:
yes, the specification says nothing about it.
but i think the implementation of trusted firmware requires that.
ref:
https://github.com/ARM-software/arm-trusted-firmware/blob/e318411f02f8a9526c97a23d0cdada2f128446e3/services/std_svc/spm/el3_spmc/spmc_shared_mem.c#L1704

Reply:
ok, it does need to be released several times, and I will modify it as soon as possible.

Reply:
Done